_posts/2021-03-14-Julia-1.6-what-has-changed-since-1.0.md

```diff
@@ -0,0 +1,737 @@
+---
+title: "Julia 1.6: what has changed since Julia 1.0?"
+layout: default
+tags:
+   - julia
+---
+
+Julia 1.0 came out well over 2 years ago.
+Since then a lot has changed and a lot hasn't.
+Since Julia 1.0 was a commitment to no breaking changes;
+but that is not to say no new features have been added to the language.
+
+<!--more-->
+
+Julia 1.6 is a huge release and it is coming out relatively soon.
+RC-1 was [released recently](https://discourse.julialang.org/t/julia-v1-6-0-rc1-is-now-available/54775).
+I suspect we have at least a few more weeks before the final release.
+The Julia Core team take a huge amount of care in not breaking any packages.
+So while that is all being checked and corner cases fixed, I think we have some time.
+Thus this post, reflecting not just on 1.6, but on everything that has happened since the 1.0 release.
+
+
+Julia 1.6 will be the first "feature" release since 1.0.
+Prior to 1.0 all releases were feature releases; they came out when they were ready.
+Since then all releases (except 1.6) have been timed releases.
+The release candidate is cut from the main branch every 3 months; and they are released after an [extensive round of additional checking](https://github.com/JuliaLang/julia/pull/35846), which often takes weeks to months.
+Julia 1.6 was soft-slated to be the new long-term-support (LTS) version that would have bug-fixes backported to it for the next few years.
+The current LTS is Julia 1.0, which has now had 5 patch releases made.
+Since it was going to be supported for a long-time people wanted to make sure everything good got in; thus it was a feature release.
+The core developers have demurred on if 1.6 will actually be selected to be the new LTS (even if it is selected, it won't ascend to being the LTS til it stops being the current Stable).
+
+My impression now is that they feel like it has too many cool new things; and that a few things didn't quite make it in even with the extended release cycle.
+So it's looking likely to me that 1.7 will actually be the LTS; but that it might also be a feature release -- possibly this time a much shorter release period than usual.
+In practice I think for a lot of package maintainers 1.6 will be a LTS, in that that is the oldest version they will make sure to continue to support.
+There have been too many cool new things (as this post will detail) to stay back to only 1.0 features.
+Already a lot of packages have dropped support for Julia versions older than 1.3.
```

Comment:
Just wondering if this might not be perceived well, "some packages" could be better only for avoiding detractors that like to pound on Julia as not being "stable"?

Reply:
Such detractors are wrong, and I think that trying to rose tint it to avoid them is not worth it.
Dropping support for old compiler versions is pretty common.
Try finding a python library supporting an ancient release